### ISSUE TITLE
Uncaught TypeError: Cannot read property 'then' of undefined

### ISSUE THREAD

**Issue by rm-code** [atom/tree-view#747]
Will try to reproduce!

**Atom Version**: 1.6.0-beta5
**System**: Mac OS X 10.11.3
**Thrown From**: [tree-view](https://github.com/atom/tree-view) package, v0.201.3

### Stack Trace

Uncaught TypeError: Cannot read property 'then' of undefined

```
At /Applications/Atom Beta.app/Contents/Resources/app.asar/node_modules/tree-view/lib/tree-view.js:477

TypeError: Cannot read property 'then' of undefined
    at TreeView.module.exports.TreeView.entryClicked (/Applications/Atom Beta.app/Contents/Resources/app.asar/node_modules/tree-view/lib/tree-view.js:477:28)
    at tree-view-file.<anonymous> (/Applications/Atom Beta.app/Contents/Resources/app.asar/node_modules/tree-view/lib/tree-view.js:183:26)
    at space-pen-div.jQuery.event.dispatch (/Applications/Atom Beta.app/Contents/Resources/app.asar/node_modules/jquery/dist/jquery.js:4435:9)
    at space-pen-div.elemData.handle (/Applications/Atom Beta.app/Contents/Resources/app.asar/node_modules/jquery/dist/jquery.js:4121:28)
```

### Commands

```
     -0:13.4.0 fuzzy-finder:toggle-file-finder (ol.tree-view.full-menu.list-tree.has-collapsable-children.focusable-panel)
     -0:12.2.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:08.9.0 editor:move-to-first-character-of-line (atom-text-editor.editor.is-focused)
     -0:01.4.0 core:close (ol.tree-view.full-menu.list-tree.has-collapsable-children.focusable-panel)
```

### Config

```json
{
  "core": {
    "followSymlinks": false,
    "packagesWithKeymapsDisabled": [
      "imdone-atom"
    ],
    "themes": [
      "base-16-ocean-dark-ui-theme",
      "base16-ocean-dark-syntax-theme"
    ]
  }
}
```

### Installed Packages

```coffee
# User
base-16-ocean-dark-ui-theme, v1.6.0
base16-ocean-dark-syntax-theme, v0.1.7
external-open, v0.6.3
file-icons, v1.6.15
language-lua, v0.9.4
language-moonscript, v1.7.1
linter, v1.11.3
linter-javac, v1.7.0
linter-luacheck, v1.1.4
linter-moonscript, v0.1.2
linter-shellcheck, v1.1.0
love-atom, v2.2.5
project-manager, v2.7.6

# Dev
No dev packages
```

**Comment by Arcanemagus** [comment]
Just a note that I hit this on v1.6.0-beta6, while double clicking on a file (that happened to already be open and focused). However it shortly afterwards stopped triggering and I have no concrete steps to reproduce :frowning:.

**Comment by m133225** [comment]
I managed to reliably reproduce this on my local machine Mac OS X 10.11.3, using atom 1.7.0-dev-107a9c6. Also downloaded and tested on atom 1.6.0-beta7 with the same results.

__Steps__
1. Add a project folder containing files if you haven't already added one.
2. Double click to open a file in the added project folders.
3. Close atom (Cmd + Q)
4. Launch atom (The file should still be open.)
5. Attempt to double click the same file from the project folder menu again.

![screen shot 2016-03-05 at 9 22 32 pm](https://cloud.githubusercontent.com/assets/12193772/13547732/6cad0f12-e318-11e5-81a4-79b6b8b9614d.png)

**Comment by BinaryMuse** [comment]
Fixed by #768; thanks all! The fix will be out in the next release.

**Comment by rm-code** [comment]
Great news :+1:
